andela/mrm_api
# #163750909 Fix soft delete bug

#### What does this PR do?
The PR ensures that the user can still create a field with the same name in the case that the older field is archived.

#### Description of Task to be completed?
Because we are not deleting rows but simply changing their state, we have scenarios where a user adding a record with say the same name as one that has already been 'archived' fails with a duplication error. The goal to ensure users can add fields even though names are similar as long as only one of them is 'active'.

#### How should this be manually tested?
- Pull the branch `bg-fix-soft-delete-163750909`
- Run migrations `make-migrate`
- run the command `make run-app`
- Run the `createLocation` query
- Run the following query


``` mutation {
  createLocation(name: "Kampala", abbreviation: "KLA", country: "Uganda", timeZone: "EAST_AFRICA_TIME", imageUrl:"https://lala.com") {   
    location {
      name
    }
  }
}
```
```
mutation{
    deleteLocation(locationId:1){
        location{
        id
        }
    }
    }
```

After the field (in this case Kampala ) has been soft-deleted,  use the `createLocation` mutation again and you should be able to create `Kampala` as a new location.

The response should be similar to that shown in the screenshots.

#### Any background context you want to provide?
 So much as one can delete and re-create a field with the same name, we cannot have two active fields with the same name. It will give an error.

#### What are the relevant pivotal tracker stories?
[#163750909](https://www.pivotaltracker.com/story/show/163750909)

#### Screenshots (if appropriate)
<img width="997" alt="screen shot 2019-02-13 at 2 52 27 pm" src="https://user-images.githubusercontent.com/22786444/52710016-55f5fa00-2f9f-11e9-862f-eeb4065d74e8.png">

#### Then the Delete
<img width="895" alt="screen shot 2019-02-13 at 2 51 48 pm" src="https://user-images.githubusercontent.com/22786444/52709977-3b238580-2f9f-11e9-982f-62db34127c70.png">

#### Then the Re-creation
<img width="997" alt="screen shot 2019-02-13 at 2 52 27 pm" src="https://user-images.githubusercontent.com/22786444/52710016-55f5fa00-2f9f-11e9-862f-eeb4065d74e8.png">

#### This  shows the response given when a user tries to create an already active existing field
<img width="971" alt="screen shot 2019-02-13 at 2 52 39 pm" src="https://user-images.githubusercontent.com/22786444/52710119-96ee0e80-2f9f-11e9-9d27-01e99df354ae.png">

#### This shows that the two fields with the same name but one is active
<img width="980" alt="screen shot 2019-02-13 at 2 58 23 pm" src="https://user-images.githubusercontent.com/22786444/52710210-d9175000-2f9f-11e9-8305-cd946530c996.png">
